cli/internal/graph/graph.go

```diff
@@ -38,51 +41,21 @@ type CompleteGraph struct {
 func (g *CompleteGraph) GetPackageTaskVisitor(ctx gocontext.Context, visitor func(ctx gocontext.Context, packageTask *nodes.PackageTask) error) func(taskID string) error {
 	return func(taskID string) error {
 		packageName, taskName := util.GetPackageTaskFromId(taskID)
-
 		pkg, ok := g.WorkspaceInfos[packageName]
 		if !ok {
 			return fmt.Errorf("cannot find package %v for task %v", packageName, taskID)
 		}
-
-		packageTask := &nodes.PackageTask{
-			TaskID:      taskID,
-			Task:        taskName,
-			PackageName: packageName,
-			Pkg:         pkg,
-		}
-
-		taskDefinition, err := g.getResolvedTaskDefinition(taskID, taskName)
-		if err != nil {
-			return err
-		}
-
-		packageTask.TaskDefinition = taskDefinition
-
-		return visitor(ctx, packageTask)
-	}
-}
-
-// getResolvedTaskDefinition currently just looks for the definition in the Pipeline
-// defined in the Graph. Later, this will get Pipelines defined in the task's workspace as well.
-func (g *CompleteGraph) getResolvedTaskDefinition(taskID string, taskName string) (*fs.TaskDefinition, error) {
-	return getTaskFromPipeline(g.Pipeline, taskID, taskName)
-}
-
-func getTaskFromPipeline(pipeline fs.Pipeline, taskID string, taskName string) (*fs.TaskDefinition, error) {
-	// first check for package-tasks
-	taskDefinition, ok := pipeline[taskID]
-	if !ok {
-		// then check for regular tasks
-		fallbackTaskDefinition, notcool := pipeline[taskName]
-		// if neither, then bail
-		if !notcool {
-			// Return an empty fs.TaskDefinition
-			return nil, fmt.Errorf("No task defined in pipeline")
+		taskDefinition, ok := g.TaskDefinitions[taskID]
```

Comment:
Instead of looking for a taskDefinition in the Pipeline, we now look for it in the `TaskDefinitions` map. The TaskDefinitions in this map will later become composed from multiple configs.